mono/metadata/class-init.c

```diff
@@ -1266,7 +1266,15 @@ make_generic_param_class (MonoGenericParam *param)
 	 * This makes sure the the value size of this class is equal to the size of the types the gparam is
 	 * constrained to, the JIT depends on this.
 	 */
-	klass->instance_size = MONO_ABI_SIZEOF (MonoObject) + mono_type_stack_size_internal (m_class_get_byval_arg (klass), NULL, TRUE);
```

Comment:
Isn't the real problem that we use `mono_type_stack_size_internal`, which should be used only to compute storage on stack and not on class layout ? And we shouldn't use it anymore at all here ? Wouldn't using `mono_type_size` just work ?

Reply:
That might work as well, there are so many size functions.

Reply:
@vargaz  Added a commit on your branch which also fixes the bugs and it looks cleaner. Let me know if it looks OK, so we can merge it for the products once the tests are finished.

Reply:
Looks ok.